Modules/LearningSequence/classes/GlobalScreen/class.ilLSToastProvider.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\GlobalScreen\Identification\IdentificationInterface;
+use ILIAS\GlobalScreen\Scope\Toast\Provider\AbstractToastProvider;
+use ILIAS\Notifications\ilNotificationOSDHandler;
+use ILIAS\Notifications\Repository\ilNotificationOSDRepository;
+
+class ilLSToastProvider extends AbstractToastProvider
+{
+    private const NOTIFICATION_TYPE = ilLSCompletionNotificationProvider::NOTIFICATION_TYPE;
+    private const NOTIFICATION_TIME_PREFERENCE_KEY = ilLSCompletionNotificationProvider::NOTIFICATION_TIME_PREFERENCE_KEY;
+
+    /**
+     * @inheritDoc
+     */
+    public function getToasts(): array
+    {
+        $toasts = [];
+        $current_user = $this->dic['ilUser'];
+        if ($current_user->getId() === 0  || $current_user->isAnonymous()) {
+            return $toasts;
+        }
+
+        $left_interval_timestamp = $current_user->getPref(self::NOTIFICATION_TIME_PREFERENCE_KEY);
+        $osd_notification_handler = new ilNotificationOSDHandler(new ilNotificationOSDRepository($this->dic->database()));
+        $notifications = $osd_notification_handler->getOSDNotificationsForUser(
+            $current_user->getId(),
+            false,
+            time() - $left_interval_timestamp,
+            self::NOTIFICATION_TYPE
+        );
+
+        if (count($notifications) === 0) {
+            return $toasts;
+        }
+
+        $factory = $this->globalScreen()->toasts()->factory();
+        $lng = $this->dic['lng'];
+        $lng->loadLanguageModule('lso');
+        $ui_factory = $this->dic['ui.factory'];
+        foreach ($notifications as $note) {
```

Comment:
This loop doesnt seem to depends on its iteration items. I would recommend 1 of 2:
- Add a specification to the toasts to present more information about the current osd.
- Just add a single toast instead of a loop.

Reply:
Thank you, good point.